### PR TITLE
Add validation functions to bindings

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

File: bindings/electron/src/index.d.ts [hunk 1]
```diff
@@ -959,3 +959,39 @@ export function testGetTestbedBootstrapOrganizationAddr(
...
+export function validateEmail(
+    raw: string
+): Promise<boolean>
+export function validateDeviceLabel(
+    raw: string
+): Promise<boolean>
+export function validateInvitationToken(
+    raw: string
+): Promise<boolean>
+export function validateBackendAddr(
```
Comment (Max-7) [inline bindings/electron/src/index.d.ts:980]: Backend addresses can already be validated using `parseBackendAddr` (with a bit more precision even, being able to tell between UserInvitation and DeviceInvitation), so there's not really any need for these functions. The frontend already uses `parseBackendAddr`, and this just adds code that will not be used and will have to be maintained.

**Review by Max-7 — APPROVED** [review]
Looks good from an API standpoint. I would remove the functions to validate backend addresses just to reduce the API size.

File: bindings/electron/src/index.d.ts (hunk 1 above)
Comment (FirelightFlagboy) [reply bindings/electron/src/index.d.ts]: The functions that check `BackendAddr*` have been removed

**Review by TimeEngineer — APPROVED** [review]

File: libparsec/src/validation.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::str::FromStr;
+
+/// Validate a label without doing a unicode normalization.
+pub fn validate_entry_name(raw: &str) -> bool {
+    libparsec_types::EntryName::is_valid(raw).is_ok()
```
Comment (touilleMan) [inline libparsec/src/validation.rs:7]: why not just `libparsec_types::EntryName::from_str(raw).is_ok()` ?

Comment (FirelightFlagboy) [reply libparsec/src/validation.rs:7]: I wanted to not have an allocation being made for validating an input, that's why I've added some `is_valid` that only take an `&str` and don't do the unicode normalization

Comment (touilleMan) [reply libparsec/src/validation.rs:7]: well shame the binding system is going to make an allocation anyway when converting the string from JS to Rust :trollface:

Comment (FirelightFlagboy) [reply libparsec/src/validation.rs:7]: I can change to use `from_str` if prefered in another PR, WDYT ?

Comment (touilleMan) [reply libparsec/src/validation.rs:7]: Don't bother that's not a big deal, just keep it in mind next time you're tempted with premature optimization ^^

**Pull request merged**